app/src/main/java/mozilla/lockbox/LockboxAutofillService.kt

```diff
@@ -82,22 +89,33 @@ class LockboxAutofillService(
             .map { latest ->
                 val state = latest.first
                 when (state) {
-                    is DataStore.State.Locked -> builder.buildAuthenticationFillResponse(this)
-                    is DataStore.State.Unlocked -> {
-                        builder.buildFilteredFillResponse(this, latest.second)
-                        ?: builder.buildFallbackFillResponse(this)
+                    is DataStore.State.Locked -> AutofillAction.Authenticate
+                    is DataStore.State.Unlocked -> AutofillAction.CompleteMultiple(latest.second)
+                    is DataStore.State.Unprepared -> AutofillAction.Cancel // we might consider onboarding here.
+                    is DataStore.State.Errored -> AutofillAction.Error(state.error)
+                }
+            }
+            .subscribe(dispatcher::dispatch)
+            .addTo(compositeDisposable)
+
+        autofillStore.autofillActions
+            .map {
+                @Suppress("IMPLICIT_CAST_TO_ANY")
+                when (it) {
+                    is AutofillAction.Complete -> builder.buildFilteredFillResponse(this, listOf(it.login)).asOptional()
```

Comment:
The Context here (`this`) is now the AutofillAction instead of the DatastoreState. Are these two interchangeable to get the package name correctly in the builder?

Reply:
`this` is the instance of the `AutofillService` in this case.

Reply:
Ahh ok. Misread that in AS :)